src/Stratis.Bitcoin.Features.MemoryPool/MempoolBehavior.cs

```diff
@@ -123,7 +124,7 @@ public class MempoolBehavior : NetworkPeerBehavior
         protected override void AttachCore()
         {
             this.AttachedPeer.MessageReceived.Register(this.OnMessageReceivedAsync);
-            this.isPeerWhitelistedForRelay = this.AttachedPeer.Behavior<IConnectionManagerBehavior>().Whitelisted && this.mempoolManager.mempoolSettings.WhiteListRelay;
+            this.isPeerWhitelistedForRelay = this.AttachedPeer.IsWhitelisted() && this.mempoolManager.mempoolSettings.WhiteListRelay;
```

Comment:
note we need to assume that `AttachedPeer` could become null in rare race conditions

Reply:
thats fine, it is handled in the extension